homework/list.js

```diff
@@ -1,79 +1,92 @@
 'use strict';
 
-(function() {
+(function(exports) {
+	var TodoListManager = function() {
+		this._listNoteContent = [];
+		this._wrapper = document.querySelector('#note-list-wrapper');
+	};
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	TodoListManager.prototype = {
+		start() {
+			this.fetchList(function(data) {		
+				this.updateList(data)
+				.then((function(){
+					this.drawList(); 
+				}).bind(this)).then((function(){
+					this.preloadFirstNote();
+				}).bind(this)).catch((function(error){
```

Comment:
Don't catch error and then silent it if you don't know what you're doing. Think about that: if an error really occur, you chain will eat it without any message. And the following chain will execute as usual. It could devastate your debugging strategy and cause lots of pain.